packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -476,18 +520,25 @@ class ConnectionStateHandler implements IConnectionStateHandler {
     // Helper method to switch between quorum and audience.
     // Old design was checking only quorum for "write" clients.
     // Latest change checks audience for all types of connections.
-    protected get membership() {
-        return this.protocol?.quorum;
+    protected get membership(): IMembership | undefined {
+        // It should not matter if we use Audience when this.readClientsWaitForJoinSignal === false.
+        // But only if it's superset of quorum, i.e. when filters to "write" clients, they are always identical!
+        // It's safer to assume that we have bugs and engaging kill-bit switch should bring us back to well-known
+        // and tested state!
```

Comment:
Took me several passes to realize what this comment was addressing.  Editorial suggestion if you want would be to start like this:

> We could always use audience here, and in practice it will probably be correct.

Then I think what you're saying is that confidence is lower on fidelity of Audience without the readClientsWaitForJoinSignal flag, so it's better to only use Audience when this is enabled. Is that right?

Reply:
Could be clearer it's used as a kill switch to actually pull into an if block `if (!flag) { return quorum } return audience`